mirrord/agent/src/file.rs

```diff
@@ -22,14 +26,15 @@ use crate::{error::Result, util::IndexAllocator};
 
 #[derive(Debug)]
 pub enum RemoteFile {
-    File(File),
+    File(File, PathBuf),
```

Comment:
Pretty sure it's not needed as you can't really call `opendir/fdopendir` on a file

Reply:
its open() then fdopendir(), the fd received by open is what we use to refer to the directory, that's how I understand it, please let me know if you have any other suggestions as to what this should be

Reply:
But if you call open on a directory you would get the directory case and not the file as it will be invalid other wise

Reply:
oh yeah my bad, we check on the metadata. ill fix this thanks!